### ISSUE TITLE
Wrong binary being downloaded (Windows binary on a Mac)

### ISSUE THREAD

**Issue by lsoares** [bonigarcia/webdrivermanager#804]
**Description of the problem**: Although I'm in a Mac, the Windows binary is being downloaded:
<img width="614" alt="image" src="https://user-images.githubusercontent.com/674045/159188873-c239cdd7-fe6e-448c-8378-e5691ba11db0.png">

**Browser and version**: Microsoft Edge 99.0.1150.39

**Operating system**: macOS Monterey 12.3

**WebDriverManager version**: 5.1.0

**WebDriverManager call**: 
```kotlin
    @BeforeAll
    fun setup() {
        WebDriverManager.edgedriver().setup()
    }

    @BeforeEach
    fun setupTest() {
        driver = EdgeDriver()
    }
```

**Error log**: 
```
/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe: /Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe: cannot execute binary file

Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
	at app//org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:576)
	at app//org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:245)
	at app//org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:161)
	at app//org.openqa.selenium.chromium.ChromiumDriver.<init>(ChromiumDriver.java:108)
	at app//org.openqa.selenium.edge.EdgeDriver.<init>(EdgeDriver.java:52)
	at app//org.openqa.selenium.edge.EdgeDriver.<init>(EdgeDriver.java:44)
	at app//org.openqa.selenium.edge.EdgeDriver.<init>(EdgeDriver.java:41)
	at app//TestAmazonCart.setupTest(TestAmazonCart.kt:20)
	at java.base@16.0.2/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@16.0.2/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base@16.0.2/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@16.0.2/java.lang.reflect.Method.invoke(Method.java:567)
	at app//org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at app//org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at app//io.qameta.allure.junit5.AllureJunit5.processFixture(AllureJunit5.java:87)
	at app//io.qameta.allure.junit5.AllureJunit5.interceptBeforeEachMethod(AllureJunit5.java:65)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.interceptLifecycleMethod(TimeoutExtension.java:126)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.interceptBeforeEachMethod(TimeoutExtension.java:76)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at app//org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeMethodInExtensionContext(ClassBasedTestDescriptor.java:506)
	at app//org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$synthesizeBeforeEachMethodAdapter$21(ClassBasedTestDescriptor.java:491)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeEachMethods$3(TestMethodTestDescriptor.java:171)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeMethodsOrCallbacksUntilExceptionOccurs$6(TestMethodTestDescriptor.java:199)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeMethodsOrCallbacksUntilExceptionOccurs(TestMethodTestDescriptor.java:199)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeEachMethods(TestMethodTestDescriptor.java:168)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base@16.0.2/java.util.ArrayList.forEach(ArrayList.java:1511)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base@16.0.2/java.util.ArrayList.forEach(ArrayList.java:1511)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at app//org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at app//org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at app//org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at app//org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base@16.0.2/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@16.0.2/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base@16.0.2/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@16.0.2/java.lang.reflect.Method.invoke(Method.java:567)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: org.openqa.selenium.WebDriverException: Driver server process died prematurely.
Build info: version: '4.1.2', revision: '9a5a329c5a'
System info: host: 'NSIT1622.lan', ip: '2001:818:d870:a300:f050:41d4:39e6:1861%en0', os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '12.3', java.version: '16.0.2'
Driver info: driver.version: EdgeDriver
	at app//org.openqa.selenium.remote.service.DriverService.start(DriverService.java:226)
	at app//org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:98)
	at app//org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:558)
	... 101 more

```

**Comment by bonigarcia** [comment]
I would need to take a look to the WebDriverManager logs (better, at `TRACE` level).

For getting the WebDriverManager logs, you need to include a Logback configuration file (for example, like [this](https://github.com/bonigarcia/webdrivermanager/blob/master/src/test/resources/logback-test.xml)) in your project classpath. The name of this file should be `src/test/resources/logback-test.xml` (if you want logs only for your tests) or `src/test/resources/logback.xml` (if you want logs for both tests and application code). Then, you need to use the following line to set the level to `TRACE`:

```
<logger name="io.github.bonigarcia" level="TRACE" />
```

After that, when running a test, your should see something like this:

```
2021-12-03 11:54:24 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.checkValidity(156) -- Removing resolution chrome=96 from cache (expired on Thu Nov 25 15:43:32 CET 2021)
2021-12-03 11:54:24 [main] TRACE i.g.b.wdm.versions.VersionDetector.getVersionsInputStream(315) -- Reading online commands.properties to find out driver version
2021-12-03 11:54:24 [main] DEBUG i.g.b.wdm.versions.VersionDetector.getBrowserVersionFromTheShell(205) -- Detecting chrome version using online commands.properties
2021-12-03 11:54:24 [main] DEBUG i.g.bonigarcia.wdm.versions.Shell.runAndWaitArray(65) -- Running command on the shell: [cmd.exe, /C, wmic, datafile, where, name="%PROGRAMFILES(X86):\=\\%\\Google\\Chrome\\Application\\chrome.exe", get, Version, /value]
2021-12-03 11:54:24 [main] DEBUG i.g.bonigarcia.wdm.versions.Shell.runAndWaitArray(69) -- Result: Version=96.0.4664.45
2021-12-03 11:54:24 [main] TRACE i.g.b.wdm.versions.VersionDetector.getBrowserVersionUsingCommand(284) -- Detected browser version is 96.0.4664.45
2021-12-03 11:54:24 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.checkValidity(156) -- Removing resolution chrome96=96.0.4664.45 from cache (expired on Thu Nov 25 18:45:50 CET 2021)
2021-12-03 11:54:25 [main] DEBUG i.g.b.wdm.versions.VersionDetector.getDriverVersionFromRepository(127) -- Latest version of chromedriver according to https://chromedriver.storage.googleapis.com/LATEST_RELEASE_96 is 96.0.4664.45
2021-12-03 11:54:25 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.resolveDriverVersion(1012) -- Using chromedriver 96.0.4664.45 (resolved driver for Chrome 96)
2021-12-03 11:54:25 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.putValueInResolutionCacheIfEmpty(119) -- Storing resolution chrome=96 in cache (valid until 12:54:25 03/12/2021 CET)
2021-12-03 11:54:25 [main] DEBUG i.g.b.wdm.cache.ResolutionCache.putValueInResolutionCacheIfEmpty(119) -- Storing resolution chrome96=96.0.4664.45 in cache (valid until 11:54:25 04/12/2021 CET)
2021-12-03 11:54:25 [main] TRACE i.g.b.wdm.cache.CacheHandler.getDriverFromCache(83) -- Checking if chromedriver exists in cache
2021-12-03 11:54:25 [main] TRACE i.g.b.wdm.cache.CacheHandler.filterCacheBy(68) -- Filter cache by chromedriver -- input list [C:\Users\boni\.cache\selenium\chromedriver\win32\81.0.4044.138\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\83.0.4103.39\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\95.0.4638.69\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\96.0.4664.45\chromedriver.exe, C:\Users\boni\.cache\selenium\geckodriver\win64\0.30.0\geckodriver.exe, C:\Users\boni\.cache\selenium\msedgedriver\win64\95.0.1020.53\msedgedriver.exe, C:\Users\boni\.cache\selenium\msedgedriver\win64\96.0.1054.29\msedgedriver.exe, C:\Users\boni\.cache\selenium\msedgedriver\win64\96.0.1054.34\msedgedriver.exe, C:\Users\boni\.cache\selenium\operadriver\win64\94.0.4606.61\operadriver.exe, C:\Users\boni\.cache\selenium\operadriver\win64\95.0.4638.54\operadriver.exe, C:\Users\boni\.cache\selenium\resolution.properties] -- output list [C:\Users\boni\.cache\selenium\chromedriver\win32\81.0.4044.138\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\83.0.4103.39\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\95.0.4638.69\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\96.0.4664.45\chromedriver.exe] 
2021-12-03 11:54:25 [main] TRACE i.g.b.wdm.cache.CacheHandler.filterCacheBy(68) -- Filter cache by 96.0.4664.45 -- input list [C:\Users\boni\.cache\selenium\chromedriver\win32\81.0.4044.138\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\83.0.4103.39\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\95.0.4638.69\chromedriver.exe, C:\Users\boni\.cache\selenium\chromedriver\win32\96.0.4664.45\chromedriver.exe] -- output list [C:\Users\boni\.cache\selenium\chromedriver\win32\96.0.4664.45\chromedriver.exe] 
2021-12-03 11:54:25 [main] TRACE i.g.b.wdm.cache.CacheHandler.filterCacheBy(68) -- Filter cache by WIN -- input list [C:\Users\boni\.cache\selenium\chromedriver\win32\96.0.4664.45\chromedriver.exe] -- output list [C:\Users\boni\.cache\selenium\chromedriver\win32\96.0.4664.45\chromedriver.exe] 
2021-12-03 11:54:25 [main] DEBUG i.g.bonigarcia.wdm.WebDriverManager.manage(968) -- Driver chromedriver 96.0.4664.45 found in cache
2021-12-03 11:54:25 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.exportDriver(1067) -- Exporting webdriver.chrome.driver as C:\Users\boni\.cache\selenium\chromedriver\win32\96.0.4664.45\chromedriver.exe
```

For further info about logging with with SLF4J and Logback you can see the following [tutorial](https://www.stubbornjava.com/posts/logging-in-java-with-slf4j-and-logback).

**Comment by lsoares** [comment]
thanks for the tip. here it is (this is what happens **before** the stack trace I posted above):

```
/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe: /Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe: cannot execute binary file
2022-03-23 13:17:57 [Test worker] DEBUG i.g.b.wdm.cache.ResolutionCache.checkKeyInResolutionCache(186) -- Resolution edge=99 in cache (valid until 14:12:16 23/03/2022 WET)
2022-03-23 13:17:57 [Test worker] DEBUG i.g.b.wdm.cache.ResolutionCache.checkKeyInResolutionCache(186) -- Resolution edge99=99.0.1150.46 in cache (valid until 13:12:16 24/03/2022 WET)
2022-03-23 13:17:57 [Test worker] INFO  i.g.bonigarcia.wdm.WebDriverManager.resolveDriverVersion(1093) -- Using msedgedriver 99.0.1150.46 (resolved driver for Edge 99)
2022-03-23 13:17:57 [Test worker] TRACE i.g.b.wdm.cache.CacheHandler.getDriverFromCache(83) -- Checking if msedgedriver exists in cache
2022-03-23 13:17:57 [Test worker] TRACE i.g.b.wdm.cache.CacheHandler.filterCacheBy(68) -- Filter cache by msedgedriver -- input list [/Users/lsoares/.cache/selenium/chromedriver/mac-arm64/99.0.4844.51/chromedriver, /Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe, /Users/lsoares/.cache/selenium/resolution.properties] -- output list [/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe] 
2022-03-23 13:17:57 [Test worker] TRACE i.g.b.wdm.cache.CacheHandler.filterCacheBy(68) -- Filter cache by 99.0.1150.46 -- input list [/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe] -- output list [/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe] 
2022-03-23 13:17:57 [Test worker] TRACE i.g.b.wdm.cache.CacheHandler.filterCacheBy(68) -- Filter cache by MAC -- input list [/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe] -- output list [/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe] 
2022-03-23 13:17:57 [Test worker] TRACE i.g.b.wdm.cache.CacheHandler.filterCacheBy(68) -- Filter cache by ARM64 -- input list [/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe] -- output list [/Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe] 
2022-03-23 13:17:57 [Test worker] DEBUG i.g.bonigarcia.wdm.WebDriverManager.manage(1049) -- Driver msedgedriver 99.0.1150.46 found in cache
2022-03-23 13:17:57 [Test worker] INFO  i.g.bonigarcia.wdm.WebDriverManager.exportDriver(1148) -- Exporting webdriver.edge.driver as /Users/lsoares/.cache/selenium/msedgedriver/mac-arm64/99.0.1150.46/msedgedriver.exe|
```

**Comment by bonigarcia** [comment]
Your drive cache is corrupt for some reason. Please run your test with:

```
WebDriverManager.edgedriver().clearDriverCache().setup();
```

... and post the logs again, please.

**Comment by lsoares** [comment]
GitHub won't let me paste here as it's too big.
[I pasted here.](https://justpaste.it/9kgp1)

**Comment by bonigarcia** [comment]
It seems there is a corner case for Edge in macOS for ARM64 architecture. I've just committed a patch to try to solve it. Unfortunately, I don't have a Mac to test it. @lsoares Could you please have try with WebDriverManager version 5.1.1-SNAPSHOT?

For that, you should to bump WebDriverManager to that version in your project and run your test again. This version (which contains the latest patches) is in the Sonatype snapshots repositories. If using Maven, you also need to include the following in your pom.xml:

```
    <repositories>
        <repository>
            <id>oss.sonatype.org-snapshot</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
            <releases>
                <enabled>false</enabled>
            </releases>
            <snapshots>
                <enabled>true</enabled>
            </snapshots>
        </repository>
    </repositories>
```

... or its equivalent in Gradle:

```
repositories {
    maven { url 'https://oss.sonatype.org/content/repositories/snapshots' }
}
```

**Comment by lsoares** [comment]
I think it is kind of fixed. now it's a different issue I believe (maybe some missing flags not sure):

https://justpaste.it/364ur

**Comment by lsoares** [comment]
I updated Edge and it works now! Thank you for the quick fix!

**Comment by bonigarcia** [comment]
@lsoares There was a problem with my last patch (related to other architectures). I changed a bit that logic and re-uploaded a new snapshot version (still 5.1.1-SNAPSHOT). Could you please run your test again? Please use:

```
WebDriverManager.edgedriver().clearDriverCache().setup();
```

**Comment by lsoares** [comment]
I did it and it still works but I don't have the previous version of Edge anymore.

**Comment by bonigarcia** [comment]
Thanks @lsoares . This fix will be available for next version.

**Comment by bonigarcia** [comment]
This is already available in WebDriverManager 5.1.1, just released.
